boto/s3/connection.py

```diff
@@ -164,19 +164,22 @@ class HostRequiredError(BotoClientError):
 class S3Connection(AWSAuthConnection):
 
     DefaultHost = 's3.amazonaws.com'
-    DefaultCallingFormat = boto.config.get('s3', 'calling_format', 'boto.s3.connection.SubdomainCallingFormat')
+    DefaultCallingFormat = 'boto.s3.connection.SubdomainCallingFormat'
     QueryString = 'Signature=%s&Expires=%d&AWSAccessKeyId=%s'
 
     def __init__(self, aws_access_key_id=None, aws_secret_access_key=None,
                  is_secure=True, port=None, proxy=None, proxy_port=None,
                  proxy_user=None, proxy_pass=None,
                  host=NoHostProvided, debug=0, https_connection_factory=None,
-                 calling_format=DefaultCallingFormat, path='/',
-                 provider='aws', bucket_class=Bucket, security_token=None,
-                 suppress_consec_slashes=True, anon=None,
-                 validate_certs=None, profile_name=None):
+                 calling_format=None, path='/', provider='aws',
+                 bucket_class=Bucket, security_token=None,
+                 suppress_consec_slashes=True, anon=None, validate_certs=None,
+                 profile_name=None):
         self.bucket_class = bucket_class
 
+        if calling_format is None:
```

Comment:
What's the benefit of transforming the class variable into an instance variable? Solves some parallelism issue?

Reply:
If we set the default at import based on the boto setting, then overrides with the -o flag don't register. That's why this needs to be in __init__.